BiconeEvolution/current_antenna_evo_build/XF_Loop/Evolutionary_Loop/Antenna_Performance_Metric/combineErrors.py

```diff
@@ -0,0 +1,89 @@
+__author__ = 'Dylan Wells <wells.1629@osu.edu>'
+
+# Imports
+import argparse
+import math
+
+import pandas as pd
+
+# Arguments
+desc = "This script will combine the errors and fitness scores \
+        of individuals made with reproduction"
+parser = argparse.ArgumentParser(description=desc)
+parser.add_argument("source", help="source directory for fitness files", type=str)
+parser.add_argument("gen", help="Generation number.", type=int)
+
+g = parser.parse_args()
+
+# load in the parents.csv file to find inviduals made from reproduction
+skiprows = 4
+parents_csv = pd.read_csv(f'{g.source}/{g.gen}_parents.csv', header=0, 
+                          delimiter=',', skiprows=skiprows)
+operators = parents_csv.iloc[:, 3].values
+
+reproduced = []
+
+for i, op in enumerate(operators):
+    if op == ' Reproduction':
+        reproduced.append(i)
+print(f"Combining errors and fitness scores of individuals {reproduced}")
+
+# load in the errors and fitnesses of parents
+fitness_csv = pd.read_csv(f'{g.source}/{(g.gen)-1}_fitnessScores.csv', header=None)
+p_fitness_scores = fitness_csv.iloc[:, 0].values
+
+errors_csv = pd.read_csv(f'{g.source}/{(g.gen)-1}_errorBars.csv', header=None)
+p_err_plus = errors_csv.iloc[:, 0].values
+p_err_minus = errors_csv.iloc[:, 1].values
+
+p_fitness_scores = [float(p_fitness_scores[i]) for i in reproduced]
+p_err_plus = [float(p_err_plus[i]) for i in reproduced]
+p_err_minus = [float(p_err_minus[i]) for i in reproduced]
+
+# load in the errors and fitnesses of children
+fitness_csv = pd.read_csv(f'{g.source}/{g.gen}_fitnessScores.csv', header=None)
+c_fitness_scores = fitness_csv.iloc[:, 0].values
+
+errors_csv = pd.read_csv(f'{g.source}/{g.gen}_errorBars.csv', header=None)
+c_err_plus = errors_csv.iloc[:, 0].values
+c_err_minus = errors_csv.iloc[:, 1].values
+
+c_fitness_scores = [float(c_fitness_scores[i]) for i in reproduced]
+c_err_plus = [float(c_err_plus[i]) for i in reproduced]
+c_err_minus = [float(c_err_minus[i]) for i in reproduced]
```

Comment:
This block seems duplicated with the one above, and they both could replaced by a function.